resources/sdk/clisdkclient/templates/restclient.mustache

```diff
@@ -0,0 +1,197 @@
+}
+
+type RESTClient struct {
+	environment string
+	token       string
+}
+
+type oAuthToken struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   string `json:"expires_in"`
+	Error       string `json:"error"`
+}
+
+//Get executes an HTTP Get
+func (r *RESTClient) Get(uri string) (string, error) {
+	return r.callAPI(http.MethodGet, uri, "")
+}
+
+//Put executes an HTTP Put
+func (r *RESTClient) Put(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPut, uri, data)
+}
+
+//Post executes an HTTP Post
+func (r *RESTClient) Post(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPost, uri, data)
+}
+
+//Patch executes an HTTP Patch
+func (r *RESTClient) Patch(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPatch, uri, data)
+}
+
+//Delete executes an HTTP Delete
+func (r *RESTClient) Delete(uri string) (string, error) {
+	return r.callAPI(http.MethodDelete, uri, "")
+}
+
+func (r *RESTClient) callAPI(method string, uri string, data string) (string, error) {
+	var apiURI *url.URL
+	if !strings.HasPrefix(uri, "/") {
+		uri = fmt.Sprintf("/%v", uri)
+	}
+	if !strings.Contains(r.environment, "localhost") {
+		apiURI, _ = url.Parse(fmt.Sprintf("https://api.%s%s", r.environment, uri))
+	} else {
+		apiURI, _ = url.Parse(fmt.Sprintf("http://%s%s", r.environment, uri))
+	}
+
+	request := &http.Request{
+		URL:    apiURI,
+		Close:  true,
+		Method: strings.ToUpper((method)),
+		Header: make(map[string][]string),
+	}
+
+	//Setting up the auth header
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", r.token))
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Cache-Control", "no-cache")
+
+    //User-Agent and SDK version headers
+    request.Header.Set("User-Agent", "{{#httpUserAgent}}{{.}}{{/httpUserAgent}}{{^httpUserAgent}}Swagger-Codegen{{/httpUserAgent}}/go-cli")
+    request.Header.Set("purecloud-sdk", "{{packageVersion}}")
+
+	if data != "" {
+		request.Body = ioutil.NopCloser(bytes.NewBuffer([]byte(data)))
+	}
+
+	//Executing the request
+	resp, err := Client.Do(request)
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	responseData := string(response)
+
+	if (resp.StatusCode != http.StatusOK) && (resp.StatusCode != http.StatusAccepted) {
+		httpError := HttpStatusError{Verb: method, Path: uri, StatusCode: resp.StatusCode, Headers: resp.Header, Body: fmt.Sprintf("%s", pretty.Pretty([]byte(responseData)))}
+		return "", httpError
+	}
+
+	return responseData, nil
+}
+
+//Login authenticates the user using the client credentials in their profile.
+func Login(config config.Configuration) (oAuthToken, error) {
```

Comment:
Can we call this `Authorize` instead of `Login`? Regardless of what name we choose for the function, the comment should use the word "authorizes" instead of "authenticates". It's pedantic, but authorization describes the act of exchanging credentials for an auth token. Authentication, which describes the act of validating a set of credentials, is only done internally within the auth service.

Reply:
I am good with that.  The problem with first cuts of names is that they are often not thought through and I like calling this authorize